src/rust/ide/view/src/project.rs

```diff
@@ -56,9 +57,30 @@ impl Model {
         Self{app,logger,display_object,graph_editor,searcher,code_editor}
     }
 
-    fn set_style(&self, is_light:bool) {
-        if is_light { self.app.themes.set_enabled(&["dark"])  }
-        else        { self.app.themes.set_enabled(&["light"]) }
+    /// Sets style of IDE.
+    pub fn set_style_internal(&self, is_light:bool) {
+        let root_elem = web::get_element_by_id("root");
+        if is_light {
+            ensogl_theme::builtin::dark::enable(&self.app);
+            match root_elem {
+                Ok(v)  => v.set_class_name("dark-theme"),
+                Err(_) => self.logger.warning("Failed to find root!"),
+            }
+        } else {
+            ensogl_theme::builtin::light::enable(&self.app);
+            match root_elem {
+                Ok(v)  => v.set_class_name("light-theme"),
+                Err(_) => self.logger.warning("Failed to find root!"),
+            }
+        }
+    }
```

Comment:
1. This is a public function with a suffix "internal".
2. This function takes bool, but neither its name nor the docs suggest what it is about.
3. A lot of repetitive code there, can be refactored.
4. Errors should not use exclamation marks.